src/main/java/ch/wisv/areafiftylan/service/TeamServiceImpl.java

```diff
@@ -1,28 +1,39 @@
 package ch.wisv.areafiftylan.service;
 
 import ch.wisv.areafiftylan.dto.TeamDTO;
+import ch.wisv.areafiftylan.dto.TeamInviteResponse;
 import ch.wisv.areafiftylan.exception.TeamNotFoundException;
+import ch.wisv.areafiftylan.exception.TokenNotFoundException;
 import ch.wisv.areafiftylan.exception.UserNotFoundException;
 import ch.wisv.areafiftylan.model.Team;
 import ch.wisv.areafiftylan.model.User;
+import ch.wisv.areafiftylan.security.TeamInviteToken;
+import ch.wisv.areafiftylan.security.Token;
 import ch.wisv.areafiftylan.service.repository.TeamRepository;
+import ch.wisv.areafiftylan.service.repository.token.TeamInviteTokenRepository;
 import com.google.common.base.Strings;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
 
-import java.util.Collection;
-import java.util.Optional;
+import javax.mail.MessagingException;
+import java.util.*;
+import java.util.stream.Collectors;
 
 @Service
 public class TeamServiceImpl implements TeamService {
     private final TeamRepository teamRepository;
     private final UserService userService;
+    private final MailService mailService;
+    private final TeamInviteTokenRepository teamInviteTokenRepository;
 
     @Autowired
-    public TeamServiceImpl(TeamRepository teamRepository, UserService userService) {
+    public TeamServiceImpl(TeamRepository teamRepository, UserService userService, MailService mailService,
+                           TeamInviteTokenRepository teamInviteTokenRepository) {
         this.teamRepository = teamRepository;
+        this.teamInviteTokenRepository = teamInviteTokenRepository;
         this.userService = userService;
+        this.mailService = mailService;
```

Comment:
Aaagh, *superautism* the the variable assignments are in a different order than the parameters and declarations.